src/helpers.js

```diff
@@ -57,10 +60,177 @@ export function idsToIgnoredRules(ruleIds = []) {
+  const config = atom.config.get('linter-eslint')
+  const hoursSinceRestart = Math.round((process.uptime() / 3600) * 10) / 10
+  let returnVal
+  try {
+    const response = await worker.request('job', {
+      type: 'debug',
+      config,
+      filePath
+    })
+    returnVal = {
+      atomVersion: atom.getVersion(),
+      linterEslintVersion: linterEslintMeta.version,
+      linterEslintConfig: config,
+      // eslint-disable-next-line import/no-dynamic-require
+      eslintVersion: require(join(response.path, 'package.json')).version,
+      hoursSinceRestart,
+      platform: process.platform,
+      eslintType: response.type,
+      eslintPath: response.path,
+    }
+  } catch (error) {
+    atom.notifications.addError(`${error}`)
+  }
+  return returnVal
+}
+
+export async function generateDebugString(worker) {
+  const debug = await getDebugInfo(worker)
+  const details = [
+    `Atom version: ${debug.atomVersion}`,
+    `linter-eslint version: ${debug.linterEslintVersion}`,
+    `ESLint version: ${debug.eslintVersion}`,
+    `Hours since last Atom restart: ${debug.hoursSinceRestart}`,
+    `Platform: ${debug.platform}`,
+    `Using ${debug.eslintType} ESLint from ${debug.eslintPath}`,
+    `linter-eslint configuration: ${JSON.stringify(debug.linterEslintConfig, null, 2)}`
+  ]
+  return details.join('\n')
+}
+
+const generateInvalidTrace = async (
+  msgLine, msgCol, msgEndLine, msgEndCol,
+  eslintFullRange, filePath, textEditor, ruleId, message, worker
+) => {
+  let errMsgRange = `${msgLine + 1}:${msgCol}`
+  if (eslintFullRange) {
+    errMsgRange += ` - ${msgEndLine + 1}:${msgEndCol + 1}`
+  }
+  const rangeText = `Requested ${eslintFullRange ? 'start point' : 'range'}: ${errMsgRange}`
+  const issueURL = 'https://github.com/AtomLinter/linter-eslint/issues/new'
+  const titleText = `Invalid position given by '${ruleId}'`
+  const title = encodeURIComponent(titleText)
+  const body = encodeURIComponent([
+    'ESLint returned a point that did not exist in the document being edited.',
+    `Rule: \`${ruleId}\``,
+    rangeText,
+    '', '',
+    '<!-- If at all possible, please include code to reproduce this issue! -->',
+    '', '',
+    'Debug information:',
+    '```json',
+    JSON.stringify(await getDebugInfo(worker), null, 2),
+    '```'
+  ].join('\n'))
+  const newIssueURL = `${issueURL}?title=${title}&body=${body}`
+  return {
+    type: 'Error',
+    severity: 'error',
+    html: `${escapeHTML(titleText)}. See the trace for details. ` +
+      `<a href="${newIssueURL}">Report this!</a>`,
+    filePath,
+    range: rangeFromLineNumber(textEditor, 0),
+    trace: [
+      {
+        type: 'Trace',
+        text: `Original message: ${ruleId} - ${message}`,
+        filePath,
+        severity: 'info',
+      },
+      {
+        type: 'Trace',
+        text: rangeText,
+        filePath,
+        severity: 'info',
+      },
+    ]
+  }
+}
+
+export async function processMessages(response, textEditor, showRule, worker) {
```

Comment:
Since this is a pretty generic function name, it might be worthwhile to add a short description of what it does in a comment.  Or better yet in a doc block, although we haven't been good about adding them up until now.